### PR TITLE
Improving UX: Intuitive User Posts and User Events Display and Dynamic Feed Functionality

### LINKED ISSUE #2196: Profile page posts don't have an intuitive interface
**Describe the bug**
- Please work with @Ayush0Chaudhary for comments, advice and review of your PR.

Profile page posts don't have an intuitive interface. The tiling method doesn't match the newsfeed layout which is unexpected.

**To Reproduce**

1. Start app
2. Click on Profile
3. Select posts

**Expected behavior**
- A post layout that better matches that of the newsfeed

**Actual behavior**
- Tiled layout

**Screenshots**
- https://www.youtube.com/watch?v=86AzK_Y_GCQ&t=505s

**Additional details**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### LINKED ISSUE #2319: Feed Synchronization Bug: Newly Posted Content Fails to Display, Pull to Refresh Ineffective
**Describe the bug**

When a user makes a post, it doesn't appear in the feed, and pulling to refresh also doesn't update the feed with the new post.

**To Reproduce**

1.Navigate to the "Create Post" section.
2.Create a new post with relevant content.
3.Submit the post.
4.Go back to the feed section.
5.Observe that the newly created post does not appear.

**Expected behavior**
The newly created post should be immediately visible in the feed at top after submission. Additionally, pulling to refresh the feed should update it with the latest posts, including the one just created at bottom.

**Actual behavior**
The post doesn't show up in the feed after submission. Pulling to refresh also does not trigger the appearance of the newly created post.

**Screenshots**

[post.webm](https://github.com/PalisadoesFoundation/talawa/assets/103507835/0b6a090d-77b5-4cce-8bce-28fe0b1eb2d3)

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Dante291**
**What kind of change does this PR introduce?**

Enhancement to the user experience by displaying user posts and user events more effectively and improving feed functionality, ensuring new posts appear prominently at the top.

**Issue Number:**

Fixes #2196 

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa/assets/103507835/21951a69-f33c-4e98-980a-8e1f21b42d43

**If relevant, did you update the documentation?**

Yes

**Summary**

Added functionality to show newly made post at very top of the feed and after refresh the old posts get at top, added User Feed and User Events Display to show Posts and Events made by user in current selected organization.

**Does this PR introduce a breaking change?**

Yes

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by Azad99-9** [comment]
Hey @Dante291, I have some suggestions with UI.
Please take them as constructive criticism.

Screenshot_20240112-004910
Screenshot_20240112-004928

1. The `placeholder text` for empty posts in above snapshots is a bit lengthy and cumbersome, you could make it a bit `crisp` or replace it with some kind of `icon` maybe.
2. The color of the text is also `Primary`, you could make it a bit lighter so that it adjusts with its background and feels more subtle & muted.

**Comment by Dante291** [comment]
@palisadoes please assign reviewers

**Comment by Dante291** [comment]
@Azad99-9 
image
 does this look better?

**Comment by Azad99-9** [comment]
@Dante291 Thanks for making requested changes.
1. For the `Icon` part it's ok.
2. The color of the icon is `vivid` and it's dominating the `text button`, please try making it bit subtle compared to text button. You could reduce its `transparency` may be.

File: lib/services/post_service.dart [hunk 1]
```diff
@@ -87,10 +97,43 @@ class PostService {
...
+  Future<void> refreshFeed() async {
+    _posts.clear();
+    _renderedPostID.clear();
+    await getPosts();
+  }
+
+  ///Method to add newly created post at the very top of the feed.
+  ///
+  /// **params**:
+  /// * `newPost`: define_the_param
```
Comment (Azad99-9) [inline lib/services/post_service.dart:116]: Please explicitly mention the `parameters` & `return types` every where in the docs, As it is a good practice.

Comment (Dante291) [reply lib/services/post_service.dart:116]: Hey Azad thanks for mentioning but I'm still working on this I will cover up all the things, proper formatting, proper docs,  proper visibility of widgets when themes are switched, and proper app localization. I need some time bro plus we need to finalize the UI too waiting for the actual volunteers to review this and share their views, so yeah give me some time....

**Comment by DMills27** [comment]
can you make the icons smaller? As it is now the protrude in the page quite a bit.

**Comment by Dante291** [comment]
> can you make the icons smaller? As it is now the protrude in the page quite a bit.

@DMills27 you mean the like and comment icons of the post? Or the empty post icon im display when there are no posts or events?

File: lib/services/post_service.dart [hunk 2]
```diff
@@ -69,7 +69,7 @@ class PostService {
   ///   None
   ///
   /// **returns**:
-  /// * `Future<void>`: define_the_return
+  /// * `Future<void>`: returns future void
```
Comment (literalEval) [inline lib/services/post_service.dart:72]: You can write None for Future<void> too. I guess ? Please let me know if that's not the case and I'll update the linter.

File: lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart [hunk 3]
```diff
@@ -35,7 +35,7 @@ class AddPostViewModel extends BaseModel {
   final TextEditingController _textHashTagController = TextEditingController();
   final TextEditingController _titleController = TextEditingController();
 
-  /// to check if in.
+  /// to check if in demo mode.
```
Comment (literalEval) [inline lib/view_model/after_auth_view_models/add_post_view_models/add_post_view_model.dart:38]: Capitalize first letter.

File: test/views/after_auth_screens/profile/user_event_test.dart [hunk 4]
```diff
@@ -125,6 +126,19 @@ void main() {
...
+      await tester.tap(find.text('Create your first event'));
+      await tester.pumpAndSettle();
+    });
+    testWidgets('check if User Event screen shows up', (tester) async {
+      when(mockViewModel.isBusy).thenReturn(false);
+      when(mockViewModel.initialise()).thenAnswer((_) async {});
+      when(mockViewModel.events).thenReturn([]);
+
+      await tester.pumpWidget(userEventsScreen(isTest: true));
+
```
Comment (literalEval) [inline test/views/after_auth_screens/profile/user_event_test.dart:138]: Remove unnecessary new lines.

**Review by literalEval — CHANGES_REQUESTED** [review]

**Comment by literalEval** [comment]
Codecov reports missing lines. Test them too. Then we'll have a final look.

File: lib/services/post_service.dart (hunk 2 above)
Comment (Dante291) [reply lib/services/post_service.dart:72]: Yes, just like how we are showing nothing for void function we do the same for future void too. If you won't mind sir, may I fix this? This way I will get to learn about custom linting practices and all.

**Comment by Dante291** [comment]
> Codecov reports missing lines. Test them too. Then we'll have a final look.

@literalEval covered all missing line except this one 

image

It should have been covered because widget is getting rendered in both app UI and testing but it doesn't impact coverage that much also it doesn't involve any logic in it. Please review

**Comment by Azad99-9** [comment]
@Dante291 use this shorthand syntax for the constructor and then rerun the test. This should solve you issue.
Screenshot from 2024-01-14 15-13-07

**Comment by Dante291** [comment]
@literalEval Please review

**Comment by Dante291** [comment]
@Ayush0Chaudhary @noman2002 @literalEval @DMills27 please review

**Comment by literalEval** [comment]
@Dante291 can you please wait for some time ? I just looked at the codebase on my local machine and realized that CLI check of custom lint reports no issues, while there are issues with docs and it shows in the IDE. Will look into your PR after fixing it.

**Review by noman2002 — CHANGES_REQUESTED** [review]
The loading indicator is getting stretched, Please fix it. And I would suggest use CupertinoActivityIndicator.

**Comment by Dante291** [comment]
> The loading indicator is getting stretched, Please fix it. And I would suggest use CupertinoActivityIndicator.

@noman2002 Sir if you're talking about the pinned post loading indicator, it is not fixed in this PR as this PR is all about implementing dynamic feed functionality and displaying user posts and events in profile of a user. 

Pinned post is not getting loaded as we are showing demo data for pinned posts for now but there's some bug that pinned posts are not getting loaded, I will make a separate issue for that and work on it.

**Review by noman2002 — APPROVED** [review]
Rest everything looks good to me.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**